src/libstd/sys/unix/fd.rs

```diff
@@ -51,12 +51,21 @@ impl FileDesc {
         Ok(ret as usize)
     }
 
+    #[cfg(not(target_env = "newlib"))]
     pub fn set_cloexec(&self) {
         unsafe {
             let ret = c::ioctl(self.fd, c::FIOCLEX);
             debug_assert_eq!(ret, 0);
         }
     }
+    #[cfg(target_env = "newlib")]
+    pub fn set_cloexec(&self) {
+        unsafe {
+            let previous = c::fnctl(self.fd, c::F_GETFD);
+            let ret = c::fnctl(self.fd, c::F_SETFD, previous | c::FD_CLOEXEC);
```

Comment:
Ah and just to confirm, newlib *doesn't* have ioctl? Looking at the recent source versions it looks like they do?